modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/pages.jsp

```diff
@@ -68,177 +68,181 @@ if (!LanguageUtil.isInheritLocales(siteGoup.getGroupId()) && !siteAdminConfigura
-		<c:if test="<%= disableLayoutSetPrototypeInput %>">
-			<div class="alert alert-info">
-				<liferay-ui:message key="you-cannot-apply-a-site-template-because-you-modified-the-display-settings-of-this-site" />
-			</div>
-		</c:if>
-
-		<aui:select disabled="<%= disableLayoutSetPrototypeInput %>" helpMessage="site-templates-with-an-incompatible-application-adapter-are-disabled" label="site-template" name="privateLayoutSetPrototypeId">
-			<aui:option label="none" selected="<%= true %>" value="" />
+			<c:choose>
+				<c:when test="<%= (publicLayoutSetPrototype != null) && !siteGoup.isStaged() && hasUnlinkLayoutSetPrototypePermission %>">
+					<c:if test="<%= disableLayoutSetPrototypeInput %>">
+						<div class="alert alert-info">
+							<liferay-ui:message key="you-cannot-enable-the-propagation-of-changes-because-you-modified-the-display-settings-of-this-site" />
+						</div>
+					</c:if>
 
-			<%
-			for (LayoutSetPrototype curLayoutSetPrototype : layoutSetPrototypes) {
-				UnicodeProperties settingsProperties = curLayoutSetPrototype.getSettingsProperties();
-			%>
+					<aui:input disabled="<%= disableLayoutSetPrototypeInput %>" inlineLabel="right" label='<%= LanguageUtil.format(request, "enable-propagation-of-changes-from-the-site-template-x", HtmlUtil.escape(publicLayoutSetPrototype.getName(locale)), false) %>' labelCssClass="simple-toggle-switch" name="publicLayoutSetPrototypeLinkEnabled" type="toggle-switch" value="<%= publicLayoutSetPrototypeLinkEnabled %>" />
 
-				<aui:option data-servletContextName='<%= settingsProperties.getProperty("customJspServletContextName", StringPool.BLANK) %>' value="<%= curLayoutSetPrototype.getLayoutSetPrototypeId() %>"><%= HtmlUtil.escape(curLayoutSetPrototype.getName(locale)) %></aui:option>
+					<div class="<%= publicLayoutSetPrototypeLinkEnabled ? "" : "hide" %>" id="<portlet:namespace />publicLayoutSetPrototypeMergeAlert">
 
-			<%
-			}
-			%>
+						<%
+						request.setAttribute("edit_layout_set_prototype.jsp-groupId", String.valueOf(siteGoup.getGroupId()));
+						request.setAttribute("edit_layout_set_prototype.jsp-layoutSet", publicLayoutSet);
+						request.setAttribute("edit_layout_set_prototype.jsp-layoutSetPrototype", publicLayoutSetPrototype);
+						request.setAttribute("edit_layout_set_prototype.jsp-redirect", currentURL);
+						%>
 
-		</aui:select>
+						<liferay-util:include page="/layout_set_merge_alert.jsp" servletContext="<%= application %>" />
+					</div>
+				</c:when>
+				<c:when test="<%= publicLayoutSetPrototype != null %>">
+					<liferay-ui:message arguments="<%= HtmlUtil.escape(publicLayoutSetPrototype.getName(locale)) %>" key="these-pages-are-linked-to-site-template-x" translateArguments="<%= false %>" />
 
-		<c:if test="<%= !siteGoup.isStaged() %>">
+					<aui:input name="publicLayoutSetPrototypeLinkEnabled" type="hidden" value="<%= publicLayoutSetPrototypeLinkEnabled %>" />
```

Comment:
This line adds a form input. Please check:

* The input is validated on the processing action

* The backend is protected agains value forging



Refer to documentation: